File: backend/src/main/java/com/backend/api/search/dto/SearchPageResponse.java
package com.backend.api.search.dto;

import lombok.*;
import org.springframework.data.domain.Page;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SearchPageResponse<T> {
    private List<T> content;
    private int currentPage;
    private int totalPages;
    private long totalElements;
    private boolean last;

    public static <T> SearchPageResponse<T> from(Page<?> page, List<T> content) {
        return SearchPageResponse.<T>builder()
                .content(content)
                .currentPage(page.getNumber() + 1)
                .totalPages(page.getTotalPages())
                .totalElements(page.getTotalElements())
                .last(page.isLast())
                .build();
    }
}

File: backend/src/main/java/com/backend/api/search/dto/SearchPageResponse.kt
package com.backend.api.search.dto

import org.springframework.data.domain.Page

data class SearchPageResponse<T>(
    val content: List<T>,
    val currentPage: Int,
    val totalPages: Int,
    val totalElements: Long,
    val last: Boolean
) {

    companion object {
        fun <T> from(page: Page<*>, content: List<T>): SearchPageResponse<T> =
            SearchPageResponse(
                content = content,
                currentPage = page.number + 1,
                totalPages = page.totalPages,
                totalElements = page.totalElements,
                last = page.isLast
            )
    }
}
